dandi/dandiapi.py

```diff
@@ -429,28 +429,33 @@ def set_raw_metadata(self, metadata: Dict[str, Any]) -> None:
             json={"metadata": metadata, "name": metadata.get("name", "")},
         )
 
-    def publish(self) -> "RemoteDandiset":
+    def wait_until_valid(self, min_time=20):
         """
-        Publish this version of the Dandiset.  Returns a copy of the
-        `RemoteDandiset` with the `version` attribute set to the new published
-        `Version`.
+        Wait for a Dandiset to be valid.  Validation is a background celery
+        task which runs asynchronously, so we need to wait for it to complete.
         """
         lgr.debug("Waiting for Dandiset %s to complete validation ...", self.identifier)
         start = time()
-        # Wait 20s for celery tasks to finish
-        while time() - start < 20:
+        while time() - start < min_time:
             r = self.client.get(f"{self.version_api_path}info/")
             if "status" not in r:
                 # Running against older version of dandi-api that doesn't
                 # validate
-                break
+                return
```

Comment:
hm, I am confused how it makes it "wait" if it returns as soon as `status` not in the returned value.

Also, note that LGTM legitimately raises a concern that `else:` below is then not reachable if all we have is `return` and not `break`

Reply:
Normally `status` is a string like `"Pending"` or `"Validating"` or `"Invalid"`. It will always be present in recent versions of dandi-api. The only time it is not present is an internal server error or a very old version of dandi-api.